### PR TITLE
completed pie stats & BE integration

### PR DESCRIPTION
**Pull request by maniknr-sureify**
completed pie stats and api integration changed the ui accordingly
image

### CODE REVIEW

File: src/components/user/ViewSinglePoll.jsx [hunk 1]
```diff
@@ -15,13 +16,15 @@ import { useLocation, useNavigate } from "react-router-dom";
 import { theme } from "../../themes/theme";
 import { StyledDuration } from "./StyledDuration";
 import axios from "axios";
-import { SAVE_POLL_URL } from "../../constants";
+import { POLL_OPTION_STATS, SAVE_POLL_URL } from "../../constants";
```
Comment (harshavardhan-sureify) [inline src/components/user/ViewSinglePoll.jsx:19]: change this to POLL_OPTION_STATS_URL

Comment (maniknr-sureify) [reply src/components/user/ViewSinglePoll.jsx:19]: done

**Review by akhilarjun3-sureify — APPROVED** [review]

**Review by harshavardhan-sureify — APPROVED** [review]

File: src/components/user/AnsweredPolls.jsx [hunk 2]
```diff
@@ -36,7 +39,17 @@ export const AnsweredPolls = () => {
         }
     };
     useEffect(() => {
-        fetchAnsweredPolls();
+        if (location.state != null) {
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/AnsweredPolls.jsx:42]: !==

File: src/components/user/ViewSinglePoll.jsx [hunk 3]
```diff
@@ -49,9 +75,16 @@ export const ViewSinglePoll = () => {
         }
         // eslint-disable-next-line
     }, []);
-
+    useEffect(() => {
+        if (poll != null) {
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/ViewSinglePoll.jsx:79]: !==

File: src/components/user/ViewSinglePoll.jsx [hunk 4]
```diff
@@ -230,12 +276,49 @@ export const ViewSinglePoll = () => {
...
+                    >
+                        <Card
+                            elevation={4}
+                            sx={{
+                                height: "80vh",
+                                maxWidth: "30rem",
+                                margin: "0px auto",
+                            }}
+                        >
+                            {pieData != null ? (
```
Comment (abhishekdundigalla-sureify) [inline src/components/user/ViewSinglePoll.jsx:297]: !==

File: src/components/admin/PollAnswers.jsx [hunk 5]
```diff
@@ -1,26 +1,19 @@
...
 import { useEffect } from "react";
-import { Box } from "@mui/material";
 import { handleToaster } from "../features/Toaster.reducer";
+import { TableComponent } from "../commonComponents/TableComponent";
+import { useState } from "react";
 
 export const PollAnswers = ({ pollId }) => {
     const [loading, setLoading] = React.useState(true);
-    const [rows, setRows] = React.useState([]);
+    const [data, setData] = useState(null);
```
Comment (abhishekdundigalla-sureify) [inline src/components/admin/PollAnswers.jsx:14]: initially, it can be useState()

File: src/components/commonComponents/PieGraph.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { ResponsivePie } from "@nivo/pie";
+export const PieGraph = ({ data }) => (
+    <ResponsivePie
```
Comment (abhishekdundigalla-sureify) [inline src/components/commonComponents/PieGraph.jsx:3]: how the pie chart is being displayed without returning it ?

File: src/components/commonComponents/TableComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    const [page, setPage] = useState(0);
+    const [rowsPerPage, setRowsPerPage] = useState(10);
+    let rows = data ?? [];
+
+    const handleChangePage = (event, newPage) => {
+        setPage(newPage);
+    };
+
+    const handleChangeRowsPerPage = (event) => {
+        setRowsPerPage(+event.target.value);
```
Comment (abhishekdundigalla-sureify) [inline src/components/commonComponents/TableComponent.jsx:29]: +event.target.value ?

File: src/components/commonComponents/PieGraph.jsx (hunk 6 above)
Comment (maniknr-sureify) [reply src/components/commonComponents/PieGraph.jsx:3]: the function automatically returns it as I haven't opened any curly braces.It is a callback function

File: src/components/commonComponents/TableComponent.jsx (hunk 7 above)
Comment (maniknr-sureify) [reply src/components/commonComponents/TableComponent.jsx:29]: It is given in Mui component like that only.

File: src/components/admin/PollAnswers.jsx (hunk 5 above)
Comment (maniknr-sureify) [reply src/components/admin/PollAnswers.jsx:14]: done

File: src/components/user/ViewSinglePoll.jsx (hunk 3 above)
Comment (maniknr-sureify) [reply src/components/user/ViewSinglePoll.jsx:79]: done

File: src/components/user/ViewSinglePoll.jsx (hunk 4 above)
Comment (maniknr-sureify) [reply src/components/user/ViewSinglePoll.jsx:297]: done

File: src/components/user/AnsweredPolls.jsx (hunk 2 above)
Comment (maniknr-sureify) [reply src/components/user/AnsweredPolls.jsx:42]: done

File: src/components/commonComponents/TableComponent.jsx (hunk 7 above)
Comment (maniknr-sureify) [reply src/components/commonComponents/TableComponent.jsx]: done

**Review by abhishekdundigalla-sureify — APPROVED** [review]

**Pull request merged**